### PR TITLE
Crsarmv7l patch 1

### PR DESCRIPTION
**Pull request by Crsarmv7l**
Fixes for 3 different functions

### CODE REVIEW

File: rflib/chipcon_nic.py [hunk 1]
```diff
@@ -763,7 +763,7 @@ def setMdmChanBW(self, bw, mhz=24, radiocfg=None):
         if chanbw_e is None:
             raise Exception("ChanBW does not translate into acceptable parameters.  Should you be changing this?")
 
-        bw = 1000.0*mhz / (8.0*(4+chanbw_m) * pow(2,chanbw_e))
```
Comment (atlas0fd00m) [inline rflib/chipcon_nic.py:766]: i'm not following, can you justify this change?  point me at a manual page or something?

File: rflib/chipcon_nic.py [hunk 2]
```diff
@@ -986,7 +986,7 @@ def calculatePktChanBW(self, mhz=24, radiocfg=None):
         #minbw = (2 * freq_uncertainty) + self.getMdmDRate() # uncertainty for both sender/receiver
         minbw = (self.getMdmDRate() + freq_uncertainty)
 
-        possibles = [ 53e3,63e3,75e3,93e3,107e3,125e3,150e3,188e3,214e3,250e3,300e3,375e3,428e3,500e3,600e3,750e3, ]
+        possibles = [63e3,75e3,93e3,107e3,125e3,150e3,188e3,214e3,250e3,300e3,375e3,428e3,500e3,600e3,750e3, ]
```
Comment (atlas0fd00m) [inline rflib/chipcon_nic.py]: is there something wrong with 53e3?

File: rflib/chipcon_nic.py [hunk 3]
```diff
@@ -965,9 +965,9 @@ def calculateMdmDeviatn(self, mhz=24, radiocfg=None):
         if baud <= 2400:
             deviatn = 5100
         elif baud <= 38400:
-            deviatn = 20000 * (old_div((baud-2400),36000))
+            deviatn = 20000 * (((baud-2400) / 36000))
         else:
-            deviatn = 129000 * (old_div((baud-38400),211600))
+            deviatn = 129000 * (((baud-38400) / 211600))
```
Comment (atlas0fd00m) [inline rflib/chipcon_nic.py:970]: these are nice

**Review by atlas0fd00m — COMMENTED** [review]
question...

**Comment by Crsarmv7l** [comment]
I can answer one of these easily, there is an issue with 53e3, minimum is
54e3 per table 63 (24mhz crystal) page 192 cc1110fx/cc1111fx datasheet.

I personally have experienced issues with 53e3 so I just removed it, should
have put in 54e3.

I'll get into the other justification later.

File: rflib/chipcon_nic.py (hunk 2 above)
Comment (Crsarmv7l) [reply rflib/chipcon_nic.py]: I responded via email, looks like it ended up at the bottom of the comment chain. Deleted and added here for relevance.

 I can answer one of these easily, there is an issue with 53e3, minimum is
54e3 per table 63 (24mhz crystal) page 192 cc1110fx/cc1111fx datasheet.

I personally have experienced issues with 53e3 so I just removed it, should
have put in 54e3. I will fix that.

File: rflib/chipcon_nic.py (hunk 1 above)
Comment (Crsarmv7l) [reply rflib/chipcon_nic.py:766]: def setMdmChanBW(self, bw, mhz=24, radiocfg=None):
        
        if radiocfg==None:
            self.getRadioConfig()
            radiocfg = self.radiocfg

        chanbw_e = None
        chanbw_m = None
        for e in range(4):
            m = int(((old_div(mhz*1000000.0, (bw *pow(2,e) * 8.0 ))) - 4) + .5)        # rounded evenly
            if m < 4:
                chanbw_e = e
                chanbw_m = m
                break
        if chanbw_e is None:
            raise Exception("ChanBW does not translate into acceptable parameters.  Should you be changing this?")

        bw = 1000.0*mhz / (8.0*(4+chanbw_m) * pow(2,chanbw_e))
        #print "chanbw_e: %x   chanbw_m: %x   chanbw: %f kHz" % (e, m, bw)

        radiocfg.mdmcfg4 &= ~(MDMCFG4_CHANBW_E | MDMCFG4_CHANBW_M)
        radiocfg.mdmcfg4 |= ((chanbw_e<<6) | (chanbw_m<<4))
        self.setRFRegister(MDMCFG4, (radiocfg.mdmcfg4))

        # from http://www.cs.jhu.edu/~carlson/download/datasheets/ask_ook_settings.pdf
        if bw > 102e3:
            self.setRFRegister(FREND1, 0xb6)
        else:
            self.setRFRegister(FREND1, 0x56)

        if bw > 325e3:
            self.setRFRegister(TEST2, 0x88)
            self.setRFRegister(TEST1, 0x31)
        else:
            self.setRFRegister(TEST2, 0x81)
            self.setRFRegister(TEST1, 0x35)
`


The function with the issue as a whole without the comments. The issue is thus: with `bw = 1000.0*mhz / (8.0*(4+chanbw_m) * pow(2,chanbw_e)) `the two following if statements:

`if bw > 102e3:
            self.setRFRegister(FREND1, 0xb6)`
and 
`if bw > 325e3:
            self.setRFRegister(TEST2, 0x88)
            self.setRFRegister(TEST1, 0x31)`

Are NEVER true.  

You can see this via the following:
chanbw_e  must be 0, 1, 2, or 3 (for e in range(4):)
chanbw_m must also be 0, 1, 2, or 3 (if m < 4:)

Given chanbw_e = 0 chanbw_m =0 and your formula:
bw = 1000.0*mhz / (8.0*(4+chanbw_m) * pow(2,chanbw_e))
bw = 750.0 

Given chanbw_e = 1 chanbw_m =1 and your formula:
bw = 1000.0*mhz / (8.0*(4+chanbw_m) * pow(2,chanbw_e))
bw = 300.0

Given chanbw_e = 2 chanbw_m =2 and your formula:
bw = 1000.0*mhz / (8.0*(4+chanbw_m) * pow(2,chanbw_e))
bw = 125.0

Given chanbw_e = 3 chanbw_m =3 and your formula:
bw = 1000.0*mhz / (8.0*(4+chanbw_m) * pow(2,chanbw_e))
bw = 53.571

(Yes I realize you can have 0 and 3 and other combos, but it is mathematically impossible for those if statements to be true with the bw formula only using 1000)

I assumed it was a typo on the bw formula given that exact same formula (but with the extra 0's added like my "fix") can be found in line 793 of chipcon_nic.py in the function getMdmChanBW, and the if statements *can* be true with 1000000.0. So you are doing it on one but not the other, both of which are ChanBW. Seen here:

Given chanbw_e = 0 chanbw_m =0 and your formula on line 793:
`bw = 1000000.0*mhz / (8.0*(4+chanbw_m) * pow(2,chanbw_e))`
bw = 750000.0

Given chanbw_e = 1 chanbw_m =1 and your formula on line 793:
`bw = 1000000.0*mhz / (8.0*(4+chanbw_m) * pow(2,chanbw_e))`
bw = 300000.0 

etc

File: rflib/chipcon_nic.py (hunk 2 above)
Comment (atlas0fd00m) [reply rflib/chipcon_nic.py]: awesome.  thanks!

File: rflib/chipcon_nic.py (hunk 1 above)
Comment (atlas0fd00m) [reply rflib/chipcon_nic.py:766]: thank you.  that was even more than i expected, and very understandable and convincing.  i also checked the manual, and realized that you're absolutely right.

**Review by atlas0fd00m — APPROVED** [review]
thanks @Crsarmv7l !

**Comment by Crsarmv7l** [comment]
> thanks @Crsarmv7l !

Thank you! Bucket list item contributing to this project. Now I just need to find a meaningful contribution to rtl_433 🤔

**Pull request merged**